django/db/migrations/operations/models.py

```diff
@@ -216,6 +216,27 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
                     related_object.field,
                     to_field,
                 )
+            # Rename M2M fields whose name is based on this model's name.
+            fields = zip(old_model._meta.local_many_to_many, new_model._meta.local_many_to_many)
```

Comment:
Your assumption here is that the order of the fields doesn't change. I think we should check for the `to` model and the `through_fields`.